megatron/model/transformer.py

```diff
@@ -699,13 +774,30 @@ def forward(self, hidden_states, attention_mask,
         # ==================================
         # core attention computation
         # ==================================
-
-        if self.checkpoint_core_attention:
-            context_layer = self._checkpointed_attention_forward(
-                query_layer, key_layer, value_layer, attention_mask, alibi)
+        if not self.use_flash_attn:
```

Comment:
This would be clearer without the double negation.

Reply:
Not really a "double" negation, but I see what you mean :p

Reply:
It is in the `else` 🤷‍♂️